### ISSUE TITLE
Segmentation fault with latest HHVM

### ISSUE THREAD

**Issue by steelbrain** [vipsoft/hhvm-ext-geoip#21]
I tried to recompile and I get this error after recompilation
```bash
steel@BlackBeauty:/var/web/PublishSpace/Async-Redis$ hhvm
Uncaught exception: Could not load extension /usr/lib/x86_64-linux-gnu/hhvm/extensions/20150212/geoip.so: getModuleBuildInfo() symbol not defined. (/usr/lib/x86_64-linux-gnu/hhvm/extensions/20150212/geoip.so: undefined symbol: getModuleBuildInfo)

Core dumped: Segmentation fault
Stack trace in /tmp/stacktrace.26174.log
Segmentation fault
```
This is the stack trace
```
# 0  ?? at hhvm:0
# 1  HPHP::geoipExtension::~geoipExtension() at /usr/lib/x86_64-linux-gnu/hhvm/extensions/20150212/geoip.so:0
# 2  __run_exit_handlers at /build/buildd/glibc-2.19/stdlib/exit.c:83
# 3  exit at /lib/x86_64-linux-gnu/libc.so.6:0
# 4  __libc_start_main at /build/buildd/glibc-2.19/csu/libc-start.c:137
# 5  ?? at hhvm:0
```

**Comment by robocoder** [comment]
Is it possible you have mixed versions of hhvm and hhvm-dev?

hhvm-ext-geoip won't even compile with the latest version of hhvm because https://github.com/vipsoft/hhvm-ext-geoip/blob/master/ext_geoip.cpp#L21 includes a file that doesn't exist anymore.

**Comment by steelbrain** [comment]
I am not sure, 'cause I am using the latest versions of hhvm-nightly and hhvm-dev.

**Comment by robocoder** [comment]
I pushed a fix in ccdcf69f. I was then able to build the extension with hhvm master HEAD, and it worked fine.

However, after I installed the hhvm-dev 3.6.0 package and rebuilt the extension, I was then able to see the segfault. I guess CMake picks up the installed version of hphp/runtime/ext/extension.h instead of the one from my hhvm source folder (i.e., CMAKE_PREFIX_PATH).

I'll leave this ticket open for further investigation...
